kubemarine/procedures/check_paas.py

```diff
@@ -217,37 +218,49 @@ def check_packages_versions(cluster: KubernetesCluster, tc: TestCase, hosts_to_p
     :param cluster: main cluster object
     :param tc: current test case object
     :param hosts_to_packages: hosts where to check packages
-    :param warn_on_bad_result: if true then uses Warning instead of Failure. Default False.
+    :param warn_on_bad_result: if true then bad results raises Warning instead of Failure. Default False.
+    :param raise_successful: if true, successful test result will be risen. Default True.
     """
     bad_results = []
+    warn_results = []
     good_results = []
 
     packages_map = pckgs.detect_installed_packages_version_hosts(cluster, hosts_to_packages)
+    packages_section = cluster.inventory['services']['packages']
+    global_cache_versions = packages_section['cache_versions']
     for package, version_map in packages_map.items():
+        package_associations = packages_section['associations'].get(cluster.get_os_family(), {}).get(
+            get_package_name(cluster.get_os_family(), package), {})
         if len(version_map) != 1:
             cluster.log.debug(f"Package {package} has different versions:")
             cluster.log.debug(version_map)
-            bad_results.append(package)
-
-        version = list(version_map.keys())[0]
-        if "not installed" in version:
-            cluster.log.debug(f"Package {package} is not installed on some nodes:")
-            cluster.log.debug(version_map[version])
-            bad_results.append(package)
-        else:
-            package_name = package.replace("*", "")
-            if version.startswith(package_name):
-                good_results.append(version)
+            if global_cache_versions and package_associations.get('cache_versions', True):
+                bad_results.append(package)
             else:
-                bad_results.append(f"Actual version: {version} doesn't match expected package version: {package}")
+                warn_results.append(package)
 
+        for version in version_map.keys():
+            if "not installed" in version:
+                cluster.log.debug(f"Package {package} is not installed on some nodes:")
+                cluster.log.debug(version_map[version])
+                bad_results.append(package)
+            else:
+                package_name = package.replace("*", "")
+                if version.startswith(package_name):
+                    good_results.append(version)
+                else:
+                    cluster.log.debug(f"Actual version: {version} doesn't match expected package version: {package} on "
+                                      f"nodes: {version_map[version]}")
+                    bad_results.append(package)
```

Comment:
Maybe add `version` instead of `package` to bad result?

Reply:
https://github.com/Netcracker/KubeMarine/pull/649/commits/ab8daf58f44c420932b99c7c785ef1926d393829